src/components/dropdown.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from "react";
+import styled, { css } from "styled-components";
+import { colors, shadows } from "../shared/constants";
+import darkArrow from "../assets/img/arrow-down-dark.svg";
+import lightArrow from "../assets/img/arrow-down-light.svg";
+import { globalTheme, themeColor } from "../shared/theme";
+import Input from "./Input";
+
+const itemStyles = css`
+  padding: 0.5rem 0;
+  background-color: ${themeColor(colors.spaceGrey, colors.white)};
+  width: 100%;
+`;
+
+const containerStyles = css`
+  box-sizing: border-box;
+  border-radius: 0.5rem;
+  transition: box-shadow 300ms ease;
+`;
+
+const DropdownContainer = styled.div`
+  ${containerStyles};
+  cursor: pointer;
+  display: flex;
+  position: relative;
+  width: 164px;
+  max-width: 200px;
+  border: solid 1px ${themeColor(colors.ikeaBlue, colors.snow)};
+  color: ${themeColor(colors.white, colors.spaceDark)};
+
+  &:hover {
+    box-shadow: ${shadows.hover};
+  }
+`;
+
+const LabelContainer = styled.div`
+  ${containerStyles};
+  ${itemStyles}
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  padding: 0.25rem 0.5rem;
+
+  & p {
+    margin: 0;
+    margin-right: 0.25rem;
+  }
+`;
+
+const iconStyling = css`
+  transition: transform 300ms ease;
+`;
+
+const IconUp = styled.img`
+  ${iconStyling};
+  transform: rotate(180deg);
+`;
+
+const IconDown = styled.img`
+  ${iconStyling};
+  transform: rotate(0deg);
+`;
+
+const ItemsContainer = styled.div`
+  ${containerStyles};
+  ${itemStyles};
+  position: absolute;
+  top: 3rem;
+  box-shadow: ${shadows.hover};
+
+  & li {
+    padding: 0.5rem 0.5rem;
+    list-style-type: none;
+
+    &:hover {
+      background-color: ${themeColor(colors.ikeaBlue, colors.snow)};
+    }
+  }
+`;
+
+interface DropdownProps {
+  label: string;
+  name: string;
+  items: string[];
+  value: string;
+  enableOther?: boolean;
+  onClick: Function;
+}
+
+const Dropdown = ({
+  name,
+  items,
+  onClick,
+  value,
+  label,
+  enableOther,
+}: DropdownProps) => {
+  const [open, setOpen] = useState(false);
+  const [other, setOther] = useState("");
+
+  useEffect(() => {
+    onClick(name, other);
+  }, [other]);
```

Comment:
This causes a (fairly significant) compiler warning which should probably be addressed.

Reply:
Fixed 🧰